lang/lowering/src/symbol_table/build.rs

```diff
@@ -101,77 +100,47 @@ impl BuildSymbolTable for Codata {
-        match symbol_table.get(name) {
-            Some(_) => {
-                return Err(LoweringError::AlreadyDefined {
-                    name: name.to_owned(),
-                    span: span.to_miette(),
-                });
-            }
-            None => {
-                let meta = DeclMeta::Dtor { params: params.clone() };
-                symbol_table.insert(name.clone(), meta);
-            }
-        }
+        check_name(symbol_table, name, span)?;
+
+        let meta = DeclMeta::Dtor { params: params.clone() };
+        symbol_table.insert(name.clone(), meta);
+
         Ok(())
     }
 }
 
 impl BuildSymbolTable for Def {
     fn build(&self, symbol_table: &mut ModuleSymbolTable) -> Result<(), LoweringError> {
         let Def { span, name, params, .. } = self;
+        check_name(symbol_table, name, span)?;
+
+        let meta = DeclMeta::Def { params: params.clone() };
+        symbol_table.insert(name.clone(), meta);
 
-        match symbol_table.get(name) {
-            Some(_) => {
-                return Err(LoweringError::AlreadyDefined {
-                    name: name.to_owned(),
-                    span: span.to_miette(),
-                });
-            }
-            None => {
-                let meta = DeclMeta::Def { params: params.clone() };
-                symbol_table.insert(name.clone(), meta);
-            }
-        }
         Ok(())
     }
 }
 
 impl BuildSymbolTable for Codef {
     fn build(&self, symbol_table: &mut ModuleSymbolTable) -> Result<(), LoweringError> {
         let Codef { span, name, params, .. } = self;
+        check_name(symbol_table, name, span)?;
+
+        let meta = DeclMeta::Codef { params: params.clone() };
+        symbol_table.insert(name.clone(), meta);
 
-        match symbol_table.get(name) {
-            Some(_) => {
-                return Err(LoweringError::AlreadyDefined {
-                    name: name.to_owned(),
-                    span: span.to_miette(),
-                });
-            }
-            None => {
-                let meta = DeclMeta::Codef { params: params.clone() };
-                symbol_table.insert(name.clone(), meta);
-            }
-        }
         Ok(())
     }
 }
 
 impl BuildSymbolTable for Let {
     fn build(&self, symbol_table: &mut ModuleSymbolTable) -> Result<(), LoweringError> {
         let Let { span, name, params, .. } = self;
-        match symbol_table.get(name) {
-            Some(_) => {
-                return Err(LoweringError::AlreadyDefined {
-                    name: name.to_owned(),
-                    span: span.to_miette(),
-                });
-            }
-            None => {
-                let meta = DeclMeta::Let { params: params.clone() };
-                symbol_table.insert(name.clone(), meta);
-            }
-        }
+        check_name(symbol_table, name, span)?;
+
+        let meta = DeclMeta::Let { params: params.clone() };
+        symbol_table.insert(name.clone(), meta);
+
```

Comment:
What about local variables? With this PR, `Type` becomes a valid local variable name, while it was a reserved keyword in all positions before.
E.g. the following program typechecks now:
```
let foo(Type: Type): Type {
   Type
}
```

Reply:
Fixed in [88189b6](https://github.com/polarity-lang/polarity/pull/440/commits/88189b68c54c76b65e9c65f71955124306c5db57)